features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/voicemessages/timeline/VoiceMessageMediaRepo.kt

```diff
@@ -88,14 +87,15 @@ class DefaultVoiceMessageMediaRepo @AssistedInject constructor(
             source = mediaSource,
             mimeType = mimeType,
             body = body,
+            useCache = false,
         ).mapCatching {
-            val dest = cachedFilePath.apply { parentFile?.mkdirs() }
-            // TODO By not closing the MediaFile we're leaking the rust file handle here.
-            // Not that big of a deal but better to avoid it someday.
-            if (it.toFile().renameTo(dest)) {
-                dest
-            } else {
-                error("Failed to move file to cache.")
+            it.use { mediaFile ->
+                val dest = cachedFilePath.apply { parentFile?.mkdirs() }
+                if (mediaFile.persist(dest.path)) {
+                    dest
+                } else {
+                    error("Failed to move file to cache.")
+                }
```

Comment:
Not from this PR, but I am discovering this file, and if `mxcUri2FilePath` throws Exception, I think the app will crash. Do you agree?

Reply:
Oh I missed it.
Yeah you're right, will fix it in another PR.

Reply:
https://github.com/vector-im/element-x-android/pull/1756

Reply:
This PR has been rebased on top of #1756 so this bug is not present here anymore.